### PR TITLE
Text separator with menu item params

### LINKED ISSUE #16198: Custom backend menu non-text separators currently un-intuitive
### Issue:
After updating to 3.7 I desperately tried to create unnamed separators in a custom backend menu. Only here on Github I found I could achieve this by naming it with a single hyphen (-). In my opinion, this is not at all intuitive. How are users supposed to know that?

What's more when named with a single hyphen, the menu item is quite hard to click in the menu item list. Of course I can select the item and use the _Edit_ button, but who the heck actually does it this way?

### Suggested solution:
I think it would be far more elegant to have a _Use Menu title as separator label_ property. And this should be switched off by default, which would be more the default behaviour of a "separator" imho. And of course the menu item type should then be named _Separator_ instead of _Text separator_
**OR:** We should have an additional _Separator_ menu item type, which doesn't even need a menu item title (no idea if this is even possible in Joomla ...) or at least works as the "classic" Joomla menu separator.

### System information
Joomla 3.7.1

### Relevant other issues
#15002 Allow menu item separators to have labels in admin menu

### PR DESCRIPTION
**Pull request by izharaazmi**
Pull Request for Issue #16198.

### Summary of Changes
Added setting to choose whether the user wants the "separator" type menu item to be displayed as standard separator (horizontal line) or labelled separator (with title text).

For B/C the titles containing only dashes and spaces will still be displayed as standard separator this can be changed in 4.0 if deemed fit. This note is included in the field description tip.

### Testing Instructions
* Apply patch
* The menu item separators should display exactly the same as before this patch. Check separators under "menu" dropdown.
* Create a new custom menu for backend and add some menu items. Also add separators between items.
* When creating separator menu items, you should be able to enter any title for it and choose whether to use title as label for the separator or not.
* Switch to this custom menu in menu module settings. Your menu should be displayed correctly with appropriate type of separators.

image

### Documentation Changes Required
Need to add information about using separator menu items in admin menu manager.

Fixes #16198.

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 8033d4159653012501b1cbdd30476dbd6d676357<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17970">issues.joomla.org/tracker/joomla-cms/17970</a>.</sub>

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 1]
```diff
@@ -132,6 +132,8 @@ COM_MENUS_ITEM_FIELD_SHOW_PAGE_HEADING_DESC="Show or hide the Browser Page Title
 COM_MENUS_ITEM_FIELD_SHOW_PAGE_HEADING_LABEL="Show Page Heading"
 COM_MENUS_ITEM_FIELD_TEMPLATE_DESC="Select a specific template style for this menu item or use the default template."
 COM_MENUS_ITEM_FIELD_TEMPLATE_LABEL="Template Style"
+COM_MENUS_ITEM_FIELD_TEXT_SEPARATOR_DESC="Choose whether this separator will display the title text as separator label. If the title contains only dashes (-) and spaces, it will not display as label."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_menus.ini]: ... as a separator label ...
... display as a label ....

Comment (izharaazmi) [reply administrator/language/en-GB/en-GB.com_menus.ini]: done!

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_menus.ini]: Looks like you only changed the first one

Comment (izharaazmi) [reply administrator/language/en-GB/en-GB.com_menus.ini]: Sorry, I misunderstood your comment. Fixed now.

**Comment by gwsdesk** [comment]
I have tested this item :white_check_mark: successfully on 7e60d1abe5aeabf0a6731b6fa85ebc2835176912<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17970">issues.joomla.org/tracker/joomla-cms/17970</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 7e60d1abe5aeabf0a6731b6fa85ebc2835176912<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17970">issues.joomla.org/tracker/joomla-cms/17970</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

File: administrator/components/com_menus/models/forms/itemadmin_separator.xml [hunk 2]
```diff
@@ -12,6 +12,21 @@
...
+		<fields name="params">
+			<field
+				name="text_separator"
+				type="radio"
+				label="COM_MENUS_ITEM_FIELD_TEXT_SEPARATOR_LABEL"
+				description="COM_MENUS_ITEM_FIELD_TEXT_SEPARATOR_DESC"
+				class="btn-group btn-group-yesno"
+				default="0"
+				filter="integer"
+			>
```
Comment (brianteeman) [inline administrator/components/com_menus/models/forms/itemadmin_separator.xml]: This needs to be indented one tab for codestyle

File: administrator/language/en-GB/en-GB.com_menus.ini [hunk 3]
```diff
@@ -132,6 +132,8 @@ COM_MENUS_ITEM_FIELD_SHOW_PAGE_HEADING_DESC="Show or hide the Browser Page Title
 COM_MENUS_ITEM_FIELD_SHOW_PAGE_HEADING_LABEL="Show Page Heading"
 COM_MENUS_ITEM_FIELD_TEMPLATE_DESC="Select a specific template style for this menu item or use the default template."
 COM_MENUS_ITEM_FIELD_TEMPLATE_LABEL="Template Style"
+COM_MENUS_ITEM_FIELD_TEXT_SEPARATOR_DESC="Choose whether this separator will be displayed as a label. If the title contains only dashes (-) and spaces, it will not be displayed as label."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_menus.ini]: displayed as label
==>
displayed as a label

**Comment by brianteeman** [comment]
@franz-wohlkoenig please remove RTC until the 2 outstanding issues are resolved

**Comment by ghost** [comment]
@brianteeman 2 new Tests? Ready to Commit-Label removed.

**Comment by brianteeman** [comment]
Just code style so no need for new tests

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by mbabker** [comment]
Merge conflicts

**Comment by izharaazmi** [comment]
@mbabker Conflicts resolved

**Pull request merged**